av/container/output.pyx

```diff
@@ -101,11 +97,7 @@ cdef class OutputContainer(Container):
             codec_context.bit_rate = 1024000
             codec_context.bit_rate_tolerance = 128000
             codec_context.ticks_per_frame = 1
-
-            rate = Fraction(rate or 24)
-
-            codec_context.framerate.num = rate.numerator
-            codec_context.framerate.den = rate.denominator
+            to_avrational(rate or 24, &codec_context.framerate)
 
             stream.avg_frame_rate = codec_context.framerate
             stream.time_base = codec_context.time_base
```

Comment:
I realize this was in the existing code, but is codec_context.time_base ever set before here?

Reply:
I *think* time_base gets initialized to a sane value, but it would be worth double checking. I really appreciate you looking at this PR, feel free to tinker and suggest changes as I'm not 100% sure of what I'm doing! When in doubt I look at how FFmpeg (CLI) does it.

Reply:
Sure, I will also tinker with it a bit when I get a chance. It doesn't seem too far off. I think some of the failing tests may actually be the result of existing mismatches between PyAV and ffmpeg. It's hard to get everything to align fully as some of the fields like framerate can get set by ffmpeg in a number of different places depending on the container/codec used.